### PR TITLE
feat(pageserver): add delta layer iterator

### PR DESCRIPTION
**Pull request by skyzh**
## Problem

part of https://github.com/neondatabase/neon/issues/8002

## Summary of changes

Add delta layer iterator and tests.

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: libs/utils/src/lsn.rs [hunk 1]
```diff
@@ -129,6 +129,8 @@ pub struct LsnParseError;
 impl Lsn {
     /// Maximum possible value for an LSN
     pub const MAX: Lsn = Lsn(u64::MAX);
+    /// Minimum possible value for an LSN
+    pub const MIN: Lsn = Lsn(u64::MIN);
```
Comment (VladLazar) [inline libs/utils/src/lsn.rs:133]: nit: I don't see this being used anywhere.

File: pageserver/src/tenant/storage_layer/delta_layer.rs [hunk 2]
```diff
@@ -1551,6 +1569,70 @@ impl<'a> pageserver_compaction::interface::CompactionDeltaEntry<'a, Key> for Del
...
+    planner: crate::tenant::vectored_blob_io::StreamingVectoredReadPlanner,
+    index_iter: crate::tenant::disk_btree::DiskBtreeIterator<'a>,
+    key_values_batch: std::collections::VecDeque<(Key, Lsn, Value)>,
+    is_end: bool,
+}
+
+#[cfg(test)]
+impl<'a> DeltaLayerIterator<'a> {
+    /// Retrieve a batch of key-value pairs into the iterator buffer.
+    async fn next_batch(&mut self) -> anyhow::Result<()> {
```
Comment (VladLazar) [inline pageserver/src/tenant/storage_layer/delta_layer.rs:1585]: nit: I'm not a big fan of `anyhow::Result`. Generally makes life more difficult down the line.

**Review by VladLazar — APPROVED** [review]
Looks good to me

File: pageserver/src/tenant/storage_layer/delta_layer.rs (hunk 2 above)
Comment (skyzh) [reply pageserver/src/tenant/storage_layer/delta_layer.rs:1585]: added a todo in my epic issue to refactor error return types in the future

File: libs/utils/src/lsn.rs (hunk 1 above)
Comment (skyzh) [reply libs/utils/src/lsn.rs]: this was from an earlier version of the patch and not used anymore with the rewrite, removed

**Pull request merged**